packages/prio3/src/prio3.ts

```diff
@@ -255,38 +294,54 @@ export class Prio3<Measurement> implements Prio3Vdaf<Measurement> {
 
     const verifier = field.decode(encodedVerifier);
 
-    let jointRand: null | Buffer = null;
+    let jointRandPart: null | Buffer = null;
     if (flp.jointRandLen > 0) {
-      [jointRand, encoded] = split(encoded, prg.seedSize);
+      [jointRandPart, encoded] = split(encoded, prg.seedSize);
     }
 
     if (encoded.length) {
       throw new Error("unused bytes at end of prepare message");
     }
 
-    return { verifier, jointRand };
+    return { verifier, jointRandPart };
   }
 
-  private encodePrepareMessage(
+  private encodePrepareShare(
     verifier: bigint[],
-    jointRandShare: Buffer | null
+    jointRandPart: Buffer | null
   ): Buffer {
     const verifierEncoded = this.field.encode(verifier);
 
-    if (this.flp.jointRandLen > 0 && jointRandShare) {
-      return Buffer.concat([verifierEncoded, jointRandShare]);
+    if (this.flp.jointRandLen > 0 && jointRandPart) {
+      return Buffer.concat([verifierEncoded, jointRandPart]);
     } else {
       return Buffer.from(verifierEncoded);
     }
   }
 
+  private decodePrepareMessage(input: Buffer): Buffer | null {
+    const { flp, prg } = this;
+
+    let jointRandCheck = null;
+    let encoded = input;
```

Comment:
Curiosity: Input could just be named `encoded`? Or is this a new variable to avoid mutating `arguments` on reassignment?